src/ParameterLists.f90

```diff
@@ -2349,6 +2352,109 @@ FUNCTION has_ParamType(thisParam,name) RESULT(hasname)
     ENDFUNCTION has_ParamType
 !
 !-------------------------------------------------------------------------------
+!> @brief This subroutine will take a parameter list of parameter lists, where
+!>        each parameter list is a column to be added in the table.  The first
+!>        parameter list must be the maximum number of rows and must be uniquely
```

Comment:
That is one of the requirements for the routine, because the first column is used to assemble the 1-D string array.  This array is then used for subsequent columns to find the row index, `i`, which is used to specify the location in the 2-D string table to assign the string.

If there a better way to get the `(i,j)` index from the parameter list, I'm open to it.

Reply:
How about searching all columns and find the maximum number of rows? Would this flexibility helpful to any of the cases using this routine?

Reply:
Not particularly.  All the tables being created have unique row names, and rather than keeping track of the indices used for each table, I have the routines here specify the row name instead.  This allows things to not be ordered for the rest of the table, assuming the first column is ordered.  It saves on some book keeping essentially.

I don't know if there would be tables that we want to create without a row label.  It is possible to make a parameter list where the first column is simply the row index integer, and when the table is passed in to be written to a file, simple do `(:,2:)` to remove it.  So there is some flexibility to do what I believe you're saying.